pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -245,7 +272,7 @@ func validatePipelineTaskName(ctx context.Context, prefix string, i int, t Pipel
 
 // validatePipelineWorkspaces validates the specified workspaces, ensuring having unique name without any empty string,
 // and validates that all the referenced workspaces (by pipeline tasks) are specified in the pipeline
-func validatePipelineWorkspaces(wss []WorkspacePipelineDeclaration, pts []PipelineTask) *apis.FieldError {
+func validatePipelineWorkspaces(wss []WorkspacePipelineDeclaration, pts []PipelineTask, finalTasks []PipelineTask) *apis.FieldError {
```

Comment:
this has the same deal as validatePipelineTasks - since finalTasks ARE PipelineTasks, i think you could pass them into this function as-is and not have to change it at all - the error message would have to be less generic but i totally think its worth it!

Reply:
its about the error message, its very easy to get lost in syntax errors with YAML and if we loose this field name, might make it even more harder 🤔 😞 I am anyways creating validation routines on `PipelineTask` as a function receiver like we talked about, will evaluate this along.